File: src/backend/api/setting.py
from typing import Any
from fastapi import APIRouter
from environment import CurrentUserDep
from utils.response_util import ResponseUtil
from service.setting_service import setting_service

setting_router = APIRouter()

# AI 接入配置
@setting_router.get("/get_coze_settings")
async def get_coze_settings(current_user: CurrentUserDep) -> Any:
    '''
    获取 Coze 的 token 以及最后更新时间
    '''
    return ResponseUtil.success(data=setting_service.get_coze_settings())

@setting_router.post("/save_coze_settings")
async def set_coze_settings(current_user: CurrentUserDep, settings: dict) -> Any:
    '''
    设置 Coze 的 token
    '''
    try:
        data = setting_service.save_coze_settings(settings)
        return ResponseUtil.success(data=data)
    except Exception as e:
        return ResponseUtil.error(msg=str(e))

@setting_router.post("/save_dify_settings")
async def save_dify_settings(current_user: CurrentUserDep, baseUrl: str) -> Any:
    '''
    设置 Dify 的地址
    '''
    try:
        data = setting_service.save_dify_settings(baseUrl)
        return ResponseUtil.success(data=data)
    except Exception as e:
        return ResponseUtil.error(msg=str(e))

@setting_router.get("/get_dify_settings")
async def get_dify_settings(current_user: CurrentUserDep) -> Any:
    '''
    获取 Dify 的地址
    '''
    try:
        data = setting_service.get_dify_settings()
        return ResponseUtil.success(data=data)
    except Exception as e:
        return ResponseUtil.error(msg=str(e))

# 智能体管理
@setting_router.post("/add_agents")
async def add_agent(current_user: CurrentUserDep, agent: dict) -> Any:
    '''
    添加智能体
    '''
    try:
        data = setting_service.add_agent(agent)
        return ResponseUtil.success(data=data)
    except Exception as e:
        return ResponseUtil.error(msg=str(e))

@setting_router.get("/get_all_agents")
async def get_all_agents(current_user: CurrentUserDep) -> Any:
    '''
    获取所有智能体
    '''
    try:
        data = setting_service.get_all_agents()
        return ResponseUtil.success(data={"agents": data})
    except Exception as e:
        return ResponseUtil.error(msg=str(e))

@setting_router.post("/save_agents")
async def save_agent(current_user: CurrentUserDep, agent: dict) -> Any:
    '''
    保存智能体
    '''
    try:
        data = setting_service.save_agent(agent)
        return ResponseUtil.success(data={"agents": data})
    except Exception as e:
        return ResponseUtil.error(msg=str(e))

@setting_router.put("/agents/{agent_id}/default")
async def set_default_agent(current_user: CurrentUserDep, agent_id: str) -> Any:
    '''
    设置默认智能体
    '''
    try:
        setting_service.set_default_agent(agent_id)
        return ResponseUtil.success()
    except Exception as e:
        return ResponseUtil.error(msg=str(e))

@setting_router.delete("/delete_agents/{agent_id}")
async def del_agent(current_user: CurrentUserDep, agent_id: str) -> Any:
    '''
    删除智能体
    '''
    try:
        setting_service.delete_agent(agent_id)
        return ResponseUtil.success(msg="删除成功")
    except Exception as e:
        return ResponseUtil.error(msg=str(e))

# 话术组管理
@setting_router.get("/get_greeting_config")
async def get_greeting_config(current_user: CurrentUserDep) -> Any:
    '''
    获取话术组配置
    '''
    try:
        data = setting_service.get_greeting_config()
        return ResponseUtil.success(data=data)
    except Exception as e:
        # 发生错误时返回空列表，避免前端崩溃
        return ResponseUtil.error(msg="Failed to load greeting config")

@setting_router.post("/save_greeting_config")
async def save_greeting_config(current_user: CurrentUserDep, config: dict) -> Any:
    '''
    保存话术组配置
    '''
    try:
        setting_service.save_greeting_config(config)
        return ResponseUtil.success()
    except Exception as e:
        return ResponseUtil.error(msg=str(e))

# 朋友圈评论
@setting_router.get("/get_moment_settings")
async def get_friend_comment_config(current_user: CurrentUserDep) -> Any:
    '''
    获取朋友圈评论配置
    '''
    try:
        data = setting_service.get_moment_settings()
        return ResponseUtil.success(data=data)
    except Exception as e:
        return ResponseUtil.error(msg=str(e))

@setting_router.post("/save_moment_settings")
async def save_friend_comment_config(current_user: CurrentUserDep, config: dict) -> Any:
    '''
    保存朋友圈评论配置
    '''
    try:
        setting_service.save_moment_settings(config)
        return ResponseUtil.success()
    except Exception as e:
        return ResponseUtil.error(msg=str(e))

# AI回复配置
@setting_router.get("/get_chat_history_settings")
async def get_ai_reply_config(current_user: CurrentUserDep) -> Any:
    '''
    获取AI回复配置
    '''
    try:
        data = setting_service.get_chat_history_settings()
        return ResponseUtil.success(data=data)
    except Exception as e:
        return ResponseUtil.error(msg=str(e))

@setting_router.post("/save_chat_history_settings")
async def save_ai_reply_config(current_user: CurrentUserDep, config: dict) -> Any:
    '''
    保存AI回复配置
    '''
    try:
        setting_service.save_chat_history_settings(config)
        return ResponseUtil.success()
    except Exception as e:
        return ResponseUtil.error(msg=str(e))

# 预警配置
@setting_router.get("/get_alert_settings")
async def get_alert_config(current_user: CurrentUserDep) -> Any:
    '''
    获取预警配置
    '''
    try:
        data = setting_service.get_alert_settings()
        return ResponseUtil.success(data=data)
    except Exception as e:
        return ResponseUtil.error(msg=str(e))

@setting_router.post("/save_alert_settings")
async def save_alert_config(current_user: CurrentUserDep, config: dict) -> Any:
    '''
    保存预警配置
    '''
    try:
        setting_service.save_alert_settings(config)
        return ResponseUtil.success()
    except Exception as e:
        return ResponseUtil.error(msg=str(e))

@setting_router.post("/test_alert_email")
async def send_test_alert_email(email: str) -> Any: 
    '''
    发送测试预警邮件
    '''
    try:
        setting_service.send_test_alert_email(email)
        return ResponseUtil.success(msg="发送邮件成功")
    except Exception as e:
        return ResponseUtil.error(msg=str(e))

# 休息时间设置
@setting_router.get("/get_rest_time_settings")
async def get_rest_time_config(current_user: CurrentUserDep) -> Any:
    '''
    获取休息时间配置
    '''
    try:
        data = setting_service.get_rest_time_settings()
        return ResponseUtil.success(data=data)
    except Exception as e:
        return ResponseUtil.error(msg=str(e))

@setting_router.post("/save_rest_time_settings")
async def save_rest_time_config(current_user: CurrentUserDep, config: dict) -> Any:
    '''
    保存休息时间配置
    '''
    try:
        setting_service.save_rest_time_settings(config)
        return ResponseUtil.success()
    except Exception as e:
        return ResponseUtil.error(msg=str(e))

File: src/frontend/src/api/setting.js
export const getGreetingConfig = async () => {
  const response = await fetch('/api/setting/greeting_config');
  if (response.ok) {
    return await response.json();
  } else {
    throw new Error('获取话术配置失败');
  }
};

export const saveGreetingConfig = async (config) => {
  const response = await fetch('/api/setting/greeting_config', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json'
    },
    body: JSON.stringify(config)
  });
  if (response.ok) {
    return await response.json();
  } else {
    throw new Error('保存话术配置失败');
  }
};

export const getDifySettings = async () => {
  const response = await fetch('/api/setting/dify_settings');
  if (response.ok) {
    return await response.json();
  } else {
    throw new Error('获取 Dify 配置失败');
  }
};

export const saveDifySettings = async (baseUrl) => {
  const response = await fetch(`/api/setting/dify_settings?baseUrl=${encodeURIComponent(baseUrl)}`, {
    method: 'POST'
  });
  if (response.ok) {
    return await response.json();
  } else {
    throw new Error('保存 Dify 配置失败');
  }
};

export const getCozeSettings = async () => {
  const response = await fetch('/api/setting/coze_settings');
  if (response.ok) {
    return await response.json();
  } else {
    throw new Error('获取 Coze 配置失败');
  }
};

export const setCozeSettings = async (token) => {
  const response = await fetch('/api/setting/coze_settings', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json'
    },
    body: JSON.stringify({ token })
  });
  if (response.ok) {
    return await response.json();
  } else {
    throw new Error('保存 Coze 配置失败');
  }
};

export const getAgents = async () => {
  const response = await fetch('/api/setting/agents');
  if (response.ok) {
    return await response.json();
  } else {
    throw new Error('获取智能体列表失败');
  }
};

export const addAgent = async (agent) => {
  const response = await fetch('/api/setting/agents', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json'
    },
    body: JSON.stringify(agent)
  });
  if (response.ok) {
    return await response.json();
  } else {
    throw new Error('添加智能体失败');
  }
};

export const deleteAgent = async (id) => {
  const response = await fetch(`/api/setting/agents/${id}`, {
    method: 'DELETE'
  });
  if (response.ok) {
    return await response.json();
  } else {
    throw new Error('删除智能体失败');
  }
};

export const setDefaultAgent = async (id) => {
  const response = await fetch(`/api/setting/agents/${id}/default`, {
    method: 'PUT'
  });
  if (response.ok) {
    return await response.json();
  } else {
    throw new Error('设置默认智能体失败');
  }
};

export const getMomentSettings = async () => {
  const response = await fetch('/api/setting/moment_settings');
  if (response.ok) {
    return await response.json();
  } else {
    throw new Error('获取朋友圈配置失败');
  }
};

export const saveMomentSettings = async (settings) => {
  const response = await fetch('/api/setting/moment_settings', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json'
    },
    body: JSON.stringify(settings)
  });
  if (response.ok) {
    return await response.json();
  } else {
    throw new Error('保存朋友圈配置失败');
  }
};

export const getChatHistorySettings = async () => {
  const response = await fetch('/api/setting/chat_history_settings');
  if (response.ok) {
    return await response.json();
  } else {
    throw new Error('获取 AI 回复配置失败');
  }
};

export const saveChatHistorySettings = async (settings) => {
  const response = await fetch('/api/setting/chat_history_settings', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json'
    },
    body: JSON.stringify(settings)
  });
  if (response.ok) {
    return await response.json();
  } else {
    throw new Error('保存 AI 回复配置失败');
  }
};

export const sendTestAlertEmail = async (email) => {
  const response = await fetch(`/api/setting/test_alert_email?email=${encodeURIComponent(email)}`, {
    method: 'POST'
  });
  if (response.ok) {
    return await response.json();
  } else {
    throw new Error('发送测试邮件失败');
  }
};

export const getAlertSettings = async () => {
  const response = await fetch('/api/setting/alert_settings');
  if (response.ok) {
    return await response.json();
  } else {
    throw new Error('获取预警配置失败');
  }
};

export const saveAlertSettings = async (settings) => {
  const response = await fetch('/api/setting/alert_settings', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json'
    },
    body: JSON.stringify(settings)
  });
  if (response.ok) {
    return await response.json();
  } else {
    throw new Error('保存预警配置失败');
  }
};

export const getRestTimeSettings = async () => {
  const response = await fetch('/api/setting/rest_time_settings');
  if (response.ok) {
    return await response.json();
  } else {
    throw new Error('获取休息时间配置失败');
  }
};

export const saveRestTimeSettings = async (settings) => {
  const response = await fetch('/api/setting/rest_time_settings', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json'
    },
    body: JSON.stringify(settings)
  });
  if (response.ok) {
    return await response.json();
  } else {
    throw new Error('保存休息时间配置失败');
  }
};
